packages/next/build/index.ts

```diff
@@ -419,9 +419,49 @@ export default async function build(dir: string, conf = null): Promise<void> {
   staticCheckWorkers.getStderr().pipe(process.stderr)
 
   const analysisBegin = process.hrtime()
+  const reservedPagesRegex = /^\/(_app|_error|_document|api)/
+  const runtimeEnvConfig = {
+    publicRuntimeConfig: config.publicRuntimeConfig,
+    serverRuntimeConfig: config.serverRuntimeConfig,
+  }
+  const distPagesDir = path.join(
+    distDir,
+    isLikeServerless
+      ? `${SERVERLESS_DIRECTORY}`
+      : `${SERVER_DIRECTORY}/static/${buildId}`,
+    `/pages`
+  )
+  const nonReservedPage = pageKeys.find(key => !key.match(reservedPagesRegex))
```

Comment:
This check seems like it gets a random page from the list of pages 🤔

Reply:
Yup, in serverless mode we just find the first non-API/reserved page and use the `_app` export from it. This is the current logic we have it's just moved up